### PR TITLE
Re-organise skiboot code

### PR DESCRIPTION
**Pull request by daxtens**
Lay out the libstb and ccan code in the same way it's laid out in skiboot.

This should make everything else in the process of getting skiboot to work as a submodule more straightforward later.

### CODE REVIEW

**Review by nick-child-ibm — CHANGES_REQUESTED** [review]
Noticed a few things. Mainly the build is failing, the notes I left should resolve this. There were also some issues during `make clean` since you added a deeper path, some of the coverage files would not delete. I recommend going into the Makefile and removing a deeper level of the coverage files with something like :
`	rm -f ./*/*/*/*/*/*.cov.* ./*/*/*/*/*.cov.* ./*/*.cov.* secvarctl-cov ./*.cov.* ./backends/*/*.cov.* ./external/*/*.cov.* ./html*
`


I would also recommend including the separation `secvar.h` into two `secvar.h` files in this PR. To refresh, when getting this project started, skiboot had two files called `secvar.h` that I needed to include. One was from skiboot `include/` and the other `libstb/secvar/`. Now that we are including more of skiboot and we have these two directories available to use, we should probably split `secvar.h` to two files, there are details in secvarctls current `secvar.h`'s comments on what code is from which `secvar.h`.

As I am writing this I am realizing there are further issues with `make clean` since some of the object files that were built are only added with conditionals on flags from the `make` command, of course `make clean` does not and should not accept flags like the build command so `rm -f $(OBJ)` will not remove the same `$(OBJ)` that was used during build. To be more specific, `crypto-openssl.o` only gets built with the `make OPENSSL=1` command, when we run `make clean` it does not delete this file since it is never added the list of object files without the `OPENSSL=1` flag. I could easily do some workaround but I am starting to dream of an easier way to just recursively delete all build files. Anyways, sorry for word dump I am thinking out loud and will post a PR soon to fix our increasingly gross cleanup. If you have any suggestions let me know. 

As always thanks for all your help and time with working on secvarctl. It is greatly appreciated.

**Comment by nick-child-ibm** [comment]
Also, not sure how closely you are trying to mirror skiboot yet but files `container_of.h` , `check_type.h` , `list.h` all live in their own sub-directory inside of the `ccan` directory in Skiboot.

File: external/skiboot/libstb/secvar/backend/edk2.h [hunk 1]
```diff
@@ -41,7 +41,7 @@
 
 #ifndef __EDK2_H__
 #define __EDK2_H__
-#include "short_types.h" //added by Nick Child
+#include "external/skiboot/include/short_types.h" //added by Nick Child, dja
```
[the remark this answers is not in the document]
Comment (daxtens) [reply external/skiboot/libstb/secvar/backend/edk2.h]: Thanks, I've added another commit which I think will help.

**Comment by daxtens** [comment]
> Also, not sure how closely you are trying to mirror skiboot yet but files `container_of.h` , `check_type.h` , `list.h` all live in their own sub-directory inside of the `ccan` directory in Skiboot.

Ah, good spot.

File: backends/edk2-compat/edk2-svc-generate.c [hunk 2]
```diff
@@ -9,9 +9,9 @@
 #include <ctype.h> // for isspace
 #include <argp.h>
 #include "crypto/crypto.h"
-#include "external/skiboot/include/endian.h"
+#include <ccan/endian/endian.h>
 #include "backends/edk2-compat/include/edk2-svc.h"
-#include "external/skiboot/include/edk2-compat-process.h" // work on factoring this out
+#include "external/skiboot/libstb/secvar/backend/edk2-compat-process.h" // work on factoring this out
```
[the remark this answers is not in the document]
Comment (daxtens) [reply backends/edk2-compat/edk2-svc-generate.c:14]: Consistency with the existing code. I haven't developed a grand theory about which includes should use `"..."` and which should use `<...>` just yet.

**Comment by daxtens** [comment]
> Noticed a few things. Mainly the build is failing, the notes I left should resolve this. There were also some issues during `make clean` since you added a deeper path, some of the coverage files would not delete. I recommend going into the Makefile and removing a deeper level of the coverage files with something like :
> ` rm -f ./*/*/*/*/*/*.cov.* ./*/*/*/*/*.cov.* ./*/*.cov.* secvarctl-cov ./*.cov.* ./backends/*/*.cov.* ./external/*/*.cov.* ./html* `

I have a patch now that uses a bigger hammer. See what you think.

> 
> I would also recommend including the separation `secvar.h` into two `secvar.h` files in this PR. To refresh, when getting this project started, skiboot had two files called `secvar.h` that I needed to include. One was from skiboot `include/` and the other `libstb/secvar/`. Now that we are including more of skiboot and we have these two directories available to use, we should probably split `secvar.h` to two files, there are details in secvarctls current `secvar.h`'s comments on what code is from which `secvar.h`.

Done.

File: Makefile [hunk 3]
```diff
@@ -68,9 +69,9 @@ secvarctl: $(OBJ)
 	$(CC) $(CFLAGS) $(_CFLAGS) -c  $< -o $@
 
 clean:
-	rm -f $(OBJ) secvarctl 
-	rm -f $(OBJ:.o=.d)
-	rm -f ./*/*.cov.* secvarctl-cov ./*.cov.* ./backends/*/*.cov.* ./external/*/*.cov.* ./html*
+	find . -name "*.[od]" -delete
+	find . -name "*.cov.*" -delete
```
Comment (nick-child-ibm) [inline Makefile:73]: `find` is included in distros right? I refuse to put it in package requirements hehehe

Comment (nick-child-ibm) [reply Makefile:73]: Regardless, this is the one-stop-cleanup-shop command I was looking for. Thanks!

**Review by nick-child-ibm — APPROVED** [review]
Everything compiles and cleans beautifully. LGTM

File: external/skiboot/libstb/secvar/secvar_util.c [hunk 4]
```diff
@@ -9,7 +9,7 @@
 #include <string.h>
 /*#include <skiboot.h>*/
 /*#include <opal.h>*/
-#include "external/skiboot/include/secvar.h"
+#include "secvar.h"
```
[the remark this answers is not in the document]
Comment (daxtens) [reply external/skiboot/libstb/secvar/secvar_util.c:12]: I am fairly sure a `"..."` include will look at the current directory first. So this should pick up `external/skiboot/libstb/secvar/secvar.h`. That's what `secvar_util.c` does in the Skiboot source: https://github.com/open-power/skiboot/blob/master/libstb/secvar/secvar_util.c#L12

**Pull request merged**